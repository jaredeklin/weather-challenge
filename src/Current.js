import React from 'react';

export const Current = ({ weatherData }) => {
  const { name, temp, description, icon, date } = weatherData;
  const imgUrl = `http://openweathermap.org/img/w/${icon}.png`;

  return (
    <section className="current-section">
      <div className="current-name">
        <h1>{name}</h1>
        <h4>{date}</h4>
      </div>
      <div className="current-info">
        <h4>{temp}</h4>
        <img src={imgUrl} alt="weather icon" />
        <p>{description}</p>
      </div>
    </section>
  );
};
