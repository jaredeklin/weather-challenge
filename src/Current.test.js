import React from 'react';
import { Current } from './Current';
import { shallow } from 'enzyme';

describe('Current', () => {
  it('should match the snapshot', () => {
    const mockProps = {
      name: 'Denver',
      temp: 70,
      imgUrl: 'www.url.com',
      description: 'clear',
      date: 'Sun'
    };

    const wrapper = shallow(<Current weatherData={mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
