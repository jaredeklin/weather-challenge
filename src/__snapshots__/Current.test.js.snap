// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Current
    weatherData={
      Object {
        "date": "Sun",
        "description": "clear",
        "imgUrl": "www.url.com",
        "name": "Denver",
        "temp": 70,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="current-name"
        >
          <h1>
            Denver
          </h1>
          <h4>
            Sun
          </h4>
        </div>,
        <div
          className="current-info"
        >
          <h4>
            70
          </h4>
          <img
            alt="weather icon"
            src="www.url.com"
          />
          <p>
            clear
          </p>
        </div>,
      ],
      "className": "current-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Denver
            </h1>,
            <h4>
              Sun
            </h4>,
          ],
          "className": "current-name",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Denver",
            },
            "ref": null,
            "rendered": "Denver",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sun",
            },
            "ref": null,
            "rendered": "Sun",
            "type": "h4",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              70
            </h4>,
            <img
              alt="weather icon"
              src="www.url.com"
            />,
            <p>
              clear
            </p>,
          ],
          "className": "current-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 70,
            },
            "ref": null,
            "rendered": 70,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "weather icon",
              "src": "www.url.com",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "clear",
            },
            "ref": null,
            "rendered": "clear",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="current-name"
          >
            <h1>
              Denver
            </h1>
            <h4>
              Sun
            </h4>
          </div>,
          <div
            className="current-info"
          >
            <h4>
              70
            </h4>
            <img
              alt="weather icon"
              src="www.url.com"
            />
            <p>
              clear
            </p>
          </div>,
        ],
        "className": "current-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Denver
              </h1>,
              <h4>
                Sun
              </h4>,
            ],
            "className": "current-name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Denver",
              },
              "ref": null,
              "rendered": "Denver",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sun",
              },
              "ref": null,
              "rendered": "Sun",
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                70
              </h4>,
              <img
                alt="weather icon"
                src="www.url.com"
              />,
              <p>
                clear
              </p>,
            ],
            "className": "current-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 70,
              },
              "ref": null,
              "rendered": 70,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "weather icon",
                "src": "www.url.com",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "clear",
              },
              "ref": null,
              "rendered": "clear",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
